mmv1/products/tpuv2/Vm.yaml

```diff
@@ -88,3 +94,220 @@ properties:
     name: 'description'
     description: |
       Text description of the TPU.
+  - !ruby/object:Api::Type::String
+    name: 'cidrBlock'
+    default_from_api: true
+    description: |
+      The CIDR block that the TPU node will use when selecting an IP address. This CIDR block must
+      be a /29 block; the Compute Engine networks API forbids a smaller block, and using a larger
+      block would be wasteful (a node can only consume one IP address). Errors will occur if the
+      CIDR block has already been used for a currently existing TPU node, the CIDR block conflicts
+      with any subnetworks in the user's provided network, or the provided network is peered with
+      another network that is using that CIDR block.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'networkConfig'
+    default_from_api: true
+    immutable: true
+    description: |
+      Network configurations for the TPU node.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'network'
+        immutable: true
+        default_from_api: true
+        description: |
+          The name of the network for the TPU node. It must be a preexisting Google Compute Engine
+          network. If none is provided, "default" will be used.
+      - !ruby/object:Api::Type::String
+        name: 'subnetwork'
+        immutable: true
+        default_from_api: true
+        description: |
+          The name of the subnetwork for the TPU node. It must be a preexisting Google Compute
+          Engine subnetwork. If none is provided, "default" will be used.
+      - !ruby/object:Api::Type::Boolean
+        name: 'enableExternalIps'
```

Comment:
Ah ok, I think we'll want the `default_value` then. I had originally added the `default_from_api` based on this diff when not specifying `network_config`:
```
             - network_config {
                  - can_ip_forward      = false -> null
                  - enable_external_ips = false -> null
                  - network             = "projects/terraform-dev-ryanoaks/global/networks/default" -> null
                  - subnetwork          = "projects/terraform-dev-ryanoaks/regions/us-central1/subnetworks/default" -> null
                }
```

Reply:
Ah- no `default_value` needed, since `false` is the default default for booleans. We should be able to delete `default_from_api`.

That was happening because the block wasn't specified, and Terraform doesn't supported nested objects natively- making the block O+C fixes that. A configuration of `network_config {}` would show the following diff without `default_from_api` on any subelements:

```
network_config {
                  - network             = "projects/terraform-dev-ryanoaks/global/networks/default" -> null
                  - subnetwork          = "projects/terraform-dev-ryanoaks/regions/us-central1/subnetworks/default" -> null
   }
```

(that's why network/subnetwork need to keep `default_from_api`, they have a complex default)

Reply:
Makes sense! I didn't have a strong understanding of some of these configs so thanks for explaining.